### PR TITLE
Fixed more than one player using a Bench/Chair

### LINKED ISSUE #1345: Multiple players can sit in a chair at once
### Version

1.4.0.0

### Bug
Multiple players can sit in a placed chair at once.

### Expected
If a chair is occupied, no other players can sit in it.

### Steps to reproduce
Cheat in blueprints for a swivel chair and build a basic habitat with it. Sit in it, and get another player to also sit in it. Both players will be sitting in the chair at once.

### PR DESCRIPTION
**Pull request by Coding-Hen**
Fixes #1345

### CODE REVIEW

**Review by dartasen — APPROVED** [review]
Looks good to me, I just have a little thing to say about consistency, those are a bit mixed up in the PR :
* `class `  =>  `public class`
* TARGET_CLASS & TARGET_METHOD => TARGET_CLASS merged into TARGET_METHOD

**Comment by Coding-Hen** [comment]
Good catch will switch them both to the same

File: NitroxPatcher/Patches/Dynamic/Bench_ExitSittingMode_Patch.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+using System.Reflection;
+using HarmonyLib;
+using NitroxClient.GameLogic;
+using NitroxClient.MonoBehaviours;
+using NitroxModel.Core;
+using NitroxModel.DataStructures;
+
+namespace NitroxPatcher.Patches.Dynamic
+{
+    class Bench_ExitSittingMode_Patch : NitroxPatch, IDynamicPatch
```
Comment (Jannify) [inline NitroxPatcher/Patches/Dynamic/Bench_ExitSittingMode_Patch.cs:11]:
```suggestion
    public class Bench_ExitSittingMode_Patch : NitroxPatch, IDynamicPatch
```

File: NitroxPatcher/Patches/Dynamic/Bench_OnHandClick_Patch.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+using NitroxClient.GameLogic.HUD;
+using NitroxClient.GameLogic.Simulation;
+using NitroxClient.MonoBehaviours;
+using NitroxModel.Core;
+using NitroxModel.DataStructures;
+using NitroxModel.Logger;
+
+namespace NitroxPatcher.Patches.Dynamic
+{
+    class Bench_OnHandClick_Patch : NitroxPatch, IDynamicPatch
```
Comment (Jannify) [inline NitroxPatcher/Patches/Dynamic/Bench_OnHandClick_Patch.cs:13]:
```suggestion
    public class Bench_OnHandClick_Patch : NitroxPatch, IDynamicPatch
```

**Review by Jannify — CHANGES_REQUESTED** [review]
This pr will just add the lock not sync the animation, am I right?

**Comment by Coding-Hen** [comment]
> 
> 
> This pr will just add the lock not sync the animation, am I right?

Yes no animation sync just locking to deny multiple people using the same chair/bench

**Review by Jannify — APPROVED** [review]

**Pull request merged**